google-pubsub/1.0/modules/ROOT/pages/google-pubsub-connector-reference.adoc

```diff
@@ -0,0 +1,641 @@
+`<pubsub:publish-message>`
+
+
+Publishes a single message.
+
+This operation is non-blocking. When batching is enabled, the message is
+stored only locally until any of the batching restrictions are reached.
+
+This operation may lead to high throughput. When using the *Ordering Key*
+field, processing messages in the order they were called is not guaranteed,
+especially in threaded environments. If you must maintain a strict order,
+set *Max Concurrency* to `1` in your source flow settings and add a delay in
+the for-each loops that contain order critical *Publish* calls.
+
+This operation returns the `messageId`, which is the unique identifier of the
+sent message in the topic.
+
+=== Parameters
+
+[%header%autowidth.spread]
+|===
+| Name | Type | Description | Default Value | Required
+| Configuration | String | Name of the configuration to use. | | x
+| Ordering Key a| String |  If not empty, the ordering key identifies related messages in which the publish order must be respected. This key is respected when the subscriber enables message ordering. |  |
+| Config Ref a| ConfigurationProvider |  Name of the configuration used to execute this component. |  | x
+| Project ID a| String |  Globally unique identifier for your project. |  |
+| Topic Name a| String |  Project-wide unique identifier for the topic. |  |
+| Message a| Binary |  Message content published to the topic. If not empty, the message must contain at least one attribute. |  |
+| Attributes a| Object |  Attributes for this message. If empty, the message must contain non-empty data. Can be used to filter messages on the subscription. |  |
+| Target Variable a| String |  Name of the variable that stores the operation's output. |  |
+| Target Value a| String |  Expression that evaluates the operation’s output. The outcome of the expression is stored in the *Target Variable* field. |  #[payload] |
+| Reconnection Strategy a| * <<Reconnect>>
+* <<ReconnectForever>> |  Retry strategy in case of connectivity errors. |  |
+|===
+
+=== Output
+
+[%autowidth.spread]
+|===
+| Type | String
+|===
+
+=== For Configurations
+
+* <<Config>>
+
+=== Throws
+
+* PUBSUB:BAD_GATEWAY
+* PUBSUB:CANCELLED
+* PUBSUB:CONNECTIVITY
+* PUBSUB:DEADLINE_EXCEEDED
+* PUBSUB:FAILED_PRECONDITION
+* PUBSUB:INVALID_ARGUMENT
+* PUBSUB:MAX_OUTSTANDING_BATCH_SIZE_REACHED
+* PUBSUB:MAX_OUTSTANDING_ELEMENT_COUNT_REACHED
+* PUBSUB:NOT_FOUND
+* PUBSUB:PERMISSION_DENIED
+* PUBSUB:RETRY_EXHAUSTED
+* PUBSUB:UNAVAILABLE
+
+
+[[SeekMessages]]
+== Seek Messages
+`<pubsub:seek-messages>`
+
+
+Supports bulk acknowledging or un-acknowledging of messages to a given snapshot or point of time based on the seekTarget provided. Seeking to a time marks every message received by Pub/Sub before the time as acknowledged, and all messages received after the time as unacknowledged. You can replay and reprocess previously acknowledged messages when seeking to a time in the past or purge messages when seeking to a time in the future.
```

Comment:
```suggestion
Supports the bulk acknowledging or un-acknowledging of messages to a given snapshot or point of time based on the provided seek target. 

Seeking to a time marks every message received by Pub/Sub before the time as acknowledged, and all messages received after the time as unacknowledged. You can replay and reprocess previously acknowledged messages when seeking to a time in the past or purge messages when seeking to a time in the future.
```

Reply:
A general comment with this material. It is complex, which makes it tough to word. It would be clearer if you separate the details that apply to seeking by snapshot and seeking to a point of time. The details should follow the order in which the features are introduced:

Supports the bulk acknowledging or un-acknowledging of messages to a point in time or to a snapshot

Seeking to a point in time...

Seeking to a snapshot enables you to...